corehq/blobs/migrate.py

```diff
@@ -305,11 +313,35 @@ def migrate(self, doc):
         return True
 
 
+class BlobDBCompressionMigrator(BlobDbMigrator):
+    def migrate(self, doc):
+        meta = doc["_obj_not_json"]
+        if meta.type_code != CODES.form_xml or meta.is_compressed:
+            return True
+        self.total_blobs += 1
+        try:
+            content = meta.open()
+        except NotFound:
+            self.save_backup(doc)
+        else:
+            with content, NamedTemporaryFile() as buffer:
+                shutil.copyfileobj(content, buffer)
+                buffer.seek(0)
```

Comment:
I'm assuming the reason for using `NamedTemporaryFile` is to work around the [`BlobStream` copy blob feature](https://github.com/dimagi/commcare-hq/blob/fc9f450ad7442e715d07a1dede61bfc887c5632c/corehq/blobs/s3db.py#L69). Would it work to use [`io.BufferedReader`](https://docs.python.org/3.7/library/io.html#io.BufferedReader) to stream directly from `content`?
```suggestion
            with content:
                buffer = BufferedReader(content)
```

Reply:
I did consider this (and I'm sure I tried it in a test) but since the blobdb is reading and writing to the same key I wanted to avoid any edge cases where the write would impact the read. I'm not sure about the `S3BlobDB` but the `FilesystemBlobDB` can't read and write to the same file concurrently.